src/components/MyProfile/ProfileTopBar/ProfileTopBar.tsx

```diff
@@ -2,20 +2,51 @@
 import React from 'react'
 import styles from './ProfileTopBar.module.scss'
 import { Button } from '@/inputs/Button/Button'
+import { useSession } from 'next-auth/react'
+import { useRouter } from 'next/router'
+import { AppRoutes } from '@/utils/routes'
 
 const ProfileTopBar = () => {
-  const handleButtonClick = () => {
-    console.log('clicked')
+  const { data: session } = useSession()
+  const userId = session?.user?.id
+
+  const router = useRouter()
+
+  const handlePublishClick = async () => {
+    if (!userId) {
+      console.error('No user ID found')
+      return
+    }
+
+    const response = await fetch(
+      `http://localhost:3000/api/profiles/${userId}/publish`,
```

Comment:
**issue:** This is wrong, as it should be profile ID not userId. If you need you can extend session object to retrieve `profileId` as well